### ISSUE TITLE
`new_term` alert broken with `query_key` and `use_terms_query`

### ISSUE THREAD

**Issue by nbrownus** [Yelp/elastalert#1868]
Recently upgraded from `v0.1.29` to `v.0.1.34` and had a few `new_term` alerts break.

The issue appears to be 2 things currently:

1. `fields` is required by the json schema validation while using only `query_key` despite the docs and how previous versions work. The error message couldn't be more unhelpful, sadly.

    ```Failed validating 'oneOf' in schema```

2. Setting both `fields` and `query_key` also fails. `fields` is a list while `query_key` is a string and the comparison will always fail https://github.com/Yelp/elastalert/blob/1ecf902afe5f49782fbbd5d569439dec07eb7d94/elastalert/ruletypes.py#L642

    ```Error initializing rule New persistence agent added: If use_terms_query is specified, you cannot specify different query_key and fields``` 

The only work around I've been successful with so far is changing the linked conditional to handle the special "list of 1" to string comparison.

Example rule:
```
alert: stdout
doc_type: test
filter:
- term:
    name: test
index: 'test'
name: test
query_key: columns.name.raw
fields: columns.name.raw
terms_window_size:
  days: 30
type: new_term
use_terms_query: true
```

**Comment by Qmando** [comment]
Thanks for reporting. I introduced this issue in #1778. I'll try to get this fixed asap.

**Comment by Qmando** [comment]
This should be fixed in 0.1.35, now available on pypi.
